### ISSUE TITLE
Add ping functionaly to tldr-bot

### ISSUE THREAD

**Issue by waldyrious** [tldr-pages/tldr#1350]
tldr-bot was first introduced in #1066, as one of the measures identified in #980 to streamline the contribution process. (The idea had been raised as early as #278.)

Besides reporting the linter errors inline for convenience, it would be nice to expand its functionality to automate as much of the maintenance process as possible, in order to avoid burn-out and prevent threads from going stale. The possibility of pinging people involved in a thread (which nicely complements the great work currently done by @mention-bot) was [mentioned in #1209](https://github.com/tldr-pages/tldr/issues/1209#issuecomment-270510219):

> In the meantime, @agnivade, do you think we could repurpose @tldr-bot to (also) perform the necessary pinging after the periods we define here expire? You've been doing that manually, but we should automate as much of this activity as possible, so as to free up contributor/maintainer time for the actual changes :)

> The bot could even close PRs as you did in #975 -- the wording you used there seems like a great template we could adopt: friendly, informative, and with clear action paths for alternative resolution.

**Comment by waldyrious** [comment]
Note: we could try the [probot-stale](https://github.com/probot/stale) integration to do this. Of course, we should be careful about closing issues/PRs, both for the reasons described [here](https://github.com/github/choosealicense.com/issues/501#issuecomment-292249558):

> I don't mind stale issues. Honestly, I like them. When a legitimate decade old bug finally gets resolved in an open source project, I'm tickled. Here, all of the ones open I intend to resolve eventually.

but especially because this could become quite aggressive if we're not careful. See [here](https://github.com/benbalter/licensee/issues/114) for example -- note particularly the:

> probot-stale bot added the `wontfix` label [and commented]:
> "This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

Lots of wrong things here: the "wontfix" label (which, even though it isn't in all caps as Bugzilla does, is still pretty aggressive), the "It will be closed if no further activity occurs" -- which, together, make the well-meaning "Thank you for your contributions." sound actually quite sarcastic. And this is evidenced by the response it got: various thumbs-down, and a reply saying

> Seriously? That's a horrible way to manage issues. In that case: bump (because you're forcing me to)

So we need to be really careful with how we implement this. Now the good news is that the bot's behavior can be [configured](https://github.com/probot/stale#usage) to avoid these issues -- taking inspiration in good examples like the aforementioned #975, and learning from bad examples like the one pointed above, #1157, [#1188](https://github.com/tldr-pages/tldr/pull/1188#issuecomment-282545800), and others I may not be remembering at the moment.




Social issues aside: using probot-stale could be preferable to expanding tldr-bot's functionality, as it would reduce the single point of failure effect, as our workflow automation integrations would be spread amongst @tldr-bot, @mention-bot, @cla-assistant and @probot-stale, but it also increases the surface area for possible failures. In any case, since tldr-bot is managed by us, we can always migrate additional functionality to it if other bots stop working.

**Comment by agnivade** [comment]
Good find. Yes, we need to be careful about the language.

I would like the bot to first ping the contributor asking for any updates. Then on further inactivity close the PR with a polite message, mentioning that it can be reopened anytime. But probot seems to be able to comment only a static message.

Let me first finish #1066 and then will look into it.

**Comment by waldyrious** [comment]
> I would like the bot to first ping the contributor asking for any updates. Then on further inactivity close the PR with a polite message, mentioning that it can be reopened anytime. But probot seems to be able to comment only a static message.

Not sure I fully understand what you mean. Probot can take two actions (first to mark an issue/PR as stale, and second to close it). From the configuration example in the README:

```yml
> # Number of days of inactivity before an Issue or Pull Request becomes stale
daysUntilStale: 60
# Number of days of inactivity before a stale Issue or Pull Request is closed
daysUntilClose: 7
```

Additionally, we can configure what messages to use in each case:

```yml
# Comment to post when marking as stale. Set to `false` to disable
markComment: >
  This issue has been automatically marked as stale because it has not had
  recent activity. It will be closed if no further activity occurs. Thank you
  for your contributions.
# Comment to post when closing a stale Issue or Pull Request. Set to `false` to disable
closeComment: false
```

So that covers what we wanted, doesn't it? Or is your concern that the message is not personalized (i.e. pinging the contributor)? If so, while I agree it's not as warm as it could be, the functional effect is the same, as the creator of a PR is always notified on comments to it.

**Comment by agnivade** [comment]
> So that covers what we wanted, doesn't it?

Oops, yes it does ! Sorry, I should have RTFM'd :sweat_smile:

**Comment by waldyrious** [comment]
Cool :) so let's see if we can agree in the timings, label names and messages to use, and look into activating probot-stale :)

**Comment by agnivade** [comment]
Great. I need to complete the finishing touches on the PR bot though.

**Comment by waldyrious** [comment]
No worries :) I made a proposal in #1360.

**Comment by waldyrious** [comment]
Ok, looks like this is fixed with #1360. There were some initial hiccups (see the four pingbacks above this comment), but apart from that, the bot seems to be able to work as intended. Closing as fixed :)
